lib/pos_blockchain.go

```diff
@@ -413,33 +410,35 @@ func (bc *Blockchain) validateBlockView(desoBlock *MsgDeSoBlock) error {
 }
 
 // validateBlockLeader validates that the proposer is the expected proposer for the
-// block height + view number pair.
-func (bc *Blockchain) validateBlockLeader(desoBlock *MsgDeSoBlock) error {
-	utxoView, err := NewUtxoView(bc.db, bc.params, bc.postgres, bc.snapshot)
+// block height + view number pair. It returns a bool indicating whether or not
+// we confirmed that the leader is invalid. If we receive an error, we are unsure
+// if the leader is invalid or not, so we return false.
+func (bc *Blockchain) validateBlockLeader(block *MsgDeSoBlock) (_isValidateFailed bool, _err error) {
```

Comment:
nit: use `isXYZ` or `hasXYZ` naming for these types of functions. For consistency with the function name, it'll be good to make the boolean return value match the true/false meaning of the function
```suggestion
func (bc *Blockchain) hasValidBlockProposerPoS(block *MsgDeSoBlock) (hasValidBlockProposer bool, _err error) {
```